ablateLibrary/boundarySolver/boundarySolver.cpp

```diff
@@ -13,6 +13,47 @@ ablate::boundarySolver::BoundarySolver::~BoundarySolver() {
     }
 }
 
+static void AddNeighborsToStencil(std::set<PetscInt>& stencilSet, DMLabel boundaryLabel, PetscInt boundaryValue, PetscInt depth, DM dm, PetscInt cell) {
+    const PetscInt maxDepth = 2;
+
+    // Check to see if this cell is already in the list
+    if (stencilSet.count(cell)) {
+        return;
+    }
+
+    // Add to the list
+    stencilSet.insert(cell);
+
+    // If not at max depth, check another layer
```

Comment:
This is a perfectly fine way of doing it. I will just note that we have https://petsc.org/main/docs/manualpages/DMPLEX/DMPlexGetAdjacency.html ,which works in a similar way.

Reply:
Thanks!  I went the more manual up/down/up approach so that I could check the face and only add cells that are on the other side of boundary faces.